compiler/src/iree/compiler/Codegen/LLVMGPU/Passes.cpp

```diff
@@ -58,19 +59,49 @@ llvm::cl::opt<int64_t> clLLVMGPUSharedMemoryLimit(
 // Bufferization Configuration
 //===----------------------------------------------------------------------===//
 
+static bool hasThreadMapping(scf::ForallOp forall) {
+  if (!forall.getMapping().has_value()) {
+    return false;
+  }
+  return llvm::any_of(forall.getMapping().value(), [](Attribute attr) {
+    return isa<gpu::GPUThreadMappingAttr>(attr);
+  });
```

Comment:
unrelated to this PR, but I've seen this code patten a bunch of time now. Maybe we should add a helper upstream for `isa` to avoid writing all these lambdas...

Reply:
that would be awesome, agree this pattern shows up quite frequently and `llvm::any_of` is relatively unwieldy.

Reply:
https://github.com/llvm/llvm-project/pull/83753